XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/HandRay.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Utilities;
+
+namespace XRTK.Providers.Controllers.Hands
+{
+    public class HandRay
```

Comment:
Shouldn't this just be an extension of a pointer?

Reply:
I disagree.

A hand ray just produces a ray that is used to raycast. Hand ray pointer should use it, but it should be possible to keep the same pointer and use different hand ray types, in order to produce different behavior.

Say you want to use this kind of hand ray instead, that doesn't emit from the palm joint.

https://twitter.com/crookookoo/status/1127621051758727169?s=19

 I think you should be able to configure the hand ray pointer, without outright replacing it.

Reply:
So in your view @provencher , what is the difference between a ray and a pointer? (as a pointer is a type of ray)
Hands may have multiple pointers for distance interaction, so why would you need a separate Ray component?

Reply:
The ray should be a part of the controller in my opinion.

A controller can point, and when you update its internal data, the ray that it can cast should be updated. This ray is simply an origin and a direction.

The pointer is a separate object that belongs to the input system, that queries the controller for the ray that is needed to cast the interactibles in the scene. It should be provided with a ray for a given hand/controller, and then draw a line between the end point and the origin.

Reply:
Eric, I can see where you're coming from but I still think there should be a clearer line between a controller, and a pointer. The pointers are the ones doing any/all raycasting, and therefore need the ray there.

Controllers only provide the location information of its current pose to the pointer, which is synchronized to the controller's pose. The focus provider is the one that actually handles the raycasting of all the registered pointers.

Controllers only have references to the pointers that are associated with them. They shouldn't have to know or care about what the pointers are doing.